oidc.go

```diff
@@ -0,0 +1,154 @@
+package ngroklistener
+
+import (
+	"errors"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+	"golang.ngrok.com/ngrok/config"
+)
+
+type oidc struct {
+	opts       []config.OIDCOption
+	OIDCOption config.HTTPEndpointOption
+
+	IssuerURL    string   `json:"issuer_url,omitempty"`
+	ClientID     string   `json:"client_id,omitempty"`
+	ClientSecret string   `json:"client_secret,omitempty"`
+	AllowEmails  []string `json:"allow_emails,omitempty"`
+	AllowDomains []string `json:"allow_domains,omitempty"`
+	Scopes       []string `json:"scopes,omitempty"`
+}
+
+func (o *oidc) Provision(caddy.Context) error {
+
+	o.doReplace()
+
+	if o.AllowEmails != nil {
+		o.opts = append(o.opts, config.WithAllowOIDCEmail(o.AllowEmails...))
+	}
+
+	if o.AllowDomains != nil {
+		o.opts = append(o.opts, config.WithAllowOIDCDomain(o.AllowDomains...))
+	}
+
+	if o.Scopes != nil {
+		o.opts = append(o.opts, config.WithOIDCScope(o.Scopes...))
+	}
+
```

Comment:
Change the check to `len(...) > 0`. See https://github.com/mohammed90/caddy-ngrok-listener/commit/8d3ac56e60cb8b5cd36ee10abb683565ee140adb for earlier unintended bug.